test/unit/notify/send-email.test.js

```diff
@@ -1,164 +1,250 @@
+  })
+
+  test('should throw error when sendEmail rejects', async () => {
+    notifyClient.prototype.sendEmail.mockRejectedValue({ Error: 'Request failed with status code 403' })
+
+    const wrapper = async () => {
+      await sendEmail(mockContext, mockMessage)
+    }
+
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw Error when sendEmail rejects', async () => {
+    notifyClient.prototype.sendEmail.mockRejectedValue({ Error: 'Request failed with status code 403' })
+
+    const wrapper = async () => {
+      await sendEmail(mockContext, mockMessage)
+    }
+
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw an error which starts with "Oh dear" when sendEmail rejects', async () => {
+    notifyClient.prototype.sendEmail.mockRejectedValue({ Error: 'Request failed with status code 403' })
+
+    const wrapper = async () => {
+      await sendEmail(mockContext, mockMessage)
+    }
+
+    expect(wrapper).rejects.toThrowError(/^Oh dear/)
+  })
+
+  test('should call notifyClient.sendEmail with correct event-notifyTemplateId when event-notifyTemplate exists in templateSchema', async () => {
+    eventTemplates = {
+      name: 'payment-request-enrichment-error',
+      notifyTemplateId: '982b92b2-da06-4c51-8996-33b13dd4ce04'
+    }
+    message.name = eventTemplates.name
+
+    await sendEmail(mockContext, message, emailAddress, mockReference)
+
+    const notifyClientMockInstance = notifyClient.mock.instances[0]
+    expect(notifyClientMockInstance.sendEmail).toHaveBeenCalledWith(eventTemplates.notifyTemplateId, emailAddress, {
+      personalisation: flatten(message),
+      reference: mockReference
+    })
+  })
+
+  test('should not call notifyClient.sendEmail with  event-notifyTemplateId when event-notifyTemplate does not exist in templateSchema', async () => {
+    eventTemplates = {
+      name: 'invalid-payment-request-enrichment-error',
+      notifyTemplateId: '982b92b2-da06-4c51-8996-33b13dd4ce04'
+    }
+    message.name = eventTemplates.name
+
+    await sendEmail(mockContext, message, emailAddress, mockReference)
+
+    const notifyClientMockInstance = notifyClient.mock.instances[0]
+    expect(notifyClientMockInstance.sendEmail).not.toHaveBeenCalledWith(eventTemplates.notifyTemplateId, emailAddress, {
+      personalisation: flatten(message),
+      reference: mockReference
+    })
+  })
+
+  test('should not call notifyClient.sendEmail with the env (general) notifyEmailTemplateId when event-notifyTemplate exists in templateSchema', async () => {
+    eventTemplates = {
+      name: 'payment-request-enrichment-error',
+      notifyTemplateId: '982b92b2-da06-4c51-8996-33b13dd4ce04'
+    }
+    message.name = eventTemplates.name
+
+    await sendEmail(mockContext, message, emailAddress, mockReference)
+
+    const notifyClientMockInstance = notifyClient.mock.instances[0]
+    expect(notifyClientMockInstance.sendEmail).not.toHaveBeenCalledWith(env.notifyEmailTemplateId, emailAddress, {

```

Comment:
you're performing `not` on this

Reply:
The `not `is on the default/general `notifyEmailTemplateId` but not `notifyTemplateId` property of the `event-notifyTemplate`  , I have modified test description to include `default` for further clarification